### PR TITLE
chore: upgrade rrweb to alpha.17

### PR DESCRIPTION
**Pull request by daibhin**
## Changes

Trying something out...

## Checklist
- [ ] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
- [ ] Accounted for the impact of any changes across different browsers
- [ ] Accounted for backwards compatibility of any changes (no breaking changes in posthog-js!)

### CODE REVIEW

**Comment by pauldambra** [comment]
we were queueing https://github.com/PostHog/posthog-js/issues/1464 on these upgrades but 🤔

**Comment by pauldambra** [comment]
i guess that this isn't working because postcss is already bundled in with rrweb, not being included by our bundler 🤔

**Comment by pauldambra** [comment]
tested locally and can still record and playback.

the bundle size increase is pretty extreme but we can loop back and improve that once we have things _fixed_

**Comment by pauldambra** [comment]
Screenshot 2024-11-02 at 23 31 05

holy cow that increase is almost all in the console recorder 🤯

**Comment by pauldambra** [comment]
🤣 🤣 🤣 🤣 🤣 🤣 🤣 🤣 🤣  @daibhin  search in the console recorder js for `postcss` 🤣 🤣 🤣 🤣 🤣 🤣 🤣 🤣 🤣

**Comment by pauldambra** [comment]
https://github.com/rrweb-io/rrweb/pull/1594

**Comment by daibhin** [comment]
> 🤣 🤣 🤣 🤣 🤣 🤣 🤣 🤣 🤣 @daibhin search in the console recorder js for `postcss` 🤣 🤣 🤣 🤣 🤣 🤣 🤣 🤣 🤣

Crikey! We might need to contribute our bundle size increase checker back to rrweb 😅 I should really never have been allowed to increase the size of that package so much in the first place

**Comment by daibhin** [comment]
@pauldambra genius!! I removed the postcss code from the console plugin and the package sizes are _finally_ back in a place of reasonable increase. Want to do a little bit of cleanup given all the changes we've made in this PR but ultimately think we're good to ship this upgrade

File: rollup.config.js [hunk 1]
```diff
@@ -81,6 +81,8 @@ const entrypointTargets = entrypoints.map((file) => {
             },
         ],
         plugins: [...pluginsForThisFile, visualizer({ filename: `bundle-stats-${fileName}.html` })],
+        // we know that recorder doesn't use postcss, but the bundler can't determine that
+        external: ['postcss'],
```
Comment (pauldambra) [inline rollup.config.js:85]: do we still need this? it didn't work, right?
```suggestion
```

File: src/__tests__/extensions/replay/mutation-rate-limiter.test.ts [hunk 2]
```diff
@@ -72,7 +71,7 @@ describe('MutationRateLimiter', () => {
 
     test('returns event if _any_ adds are left', () => {
         const event = makeEvent({
-            adds: [{ parentId: 0, nextId: 0, node: {} as unknown as serializedNodeWithId }],
```
Comment (daibhin) [inline src/__tests__/extensions/replay/mutation-rate-limiter.test.ts:75]: Will have to come back and type this once https://github.com/rrweb-io/rrweb/pull/1593 merges

**Review by pauldambra — APPROVED** [review]
works locally for me....

**Pull request merged**